packages/flutter/test/material/icon_button_test.dart

```diff
@@ -44,6 +44,26 @@ void main() {
     expect(mockOnPressedFunction.called, 1);
   });
 
+  testWidgets('test icon respects icon theme size', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      wrap(
+        child: Theme(
+          data: ThemeData(iconTheme: const IconThemeData(size: 10)),
+          child: IconButton(
+            onPressed: mockOnPressedFunction.handler,
+            icon: const Icon(Icons.link),
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox iconButton = tester.renderObject(find.byType(IconButton));
+    expect(iconButton.size, const Size(48.0, 48.0));
```

Comment:
I am not sure I am missing something. The size provided by the theme is 10, should this not be 10?

Reply:
the size of 10 was poorly chosen, as the Icon size gets bumped to 48 if is its under a certain value. This is true regardless if it is set directly or through a theme. 

I changed it to 80 without padding so that the size of the RenderBox is now also 80.